File: web/src/api/contract.ts
import request from '../utils/request'

export interface Contract {
  id: number
  tenant_id: number
  title: string
  party_a: string
  party_b: string
  amount: number
  start_date: string
  end_date: string
  status: 'draft' | 'active' | 'expired' | 'terminated'
  file_url: string
  remark: string
  created_at: string
  updated_at: string
}

export function listContracts(page = 1, page_size = 20) {
  return request.get('/contracts', { params: { page, page_size } })
}

export function getContract(id: number) {
  return request.get(`/contracts/${id}`)
}

export function createContract(data: Partial<Contract>) {
  return request.post('/contracts', data)
}

export function updateContract(id: number, data: Partial<Contract>) {
  return request.put(`/contracts/${id}`, data)
}

export function deleteContract(id: number) {
  return request.delete(`/contracts/${id}`)
}

File: server/ent/schema/contract.go
package schema

import (
	"time"

	"entgo.io/ent"
	"entgo.io/ent/schema/edge"
	"entgo.io/ent/schema/field"
	"entgo.io/ent/schema/index"
)

// Contract holds the schema definition for the Contract entity.
type Contract struct {
	ent.Schema
}

// Mixin of the Contract.
func (Contract) Mixin() []ent.Mixin {
	return []ent.Mixin{
		TimeMixin{},
	}
}

// Fields of the Contract.
func (Contract) Fields() []ent.Field {
	return []ent.Field{
		field.Int("tenant_id").Comment("所属租户ID"),
		field.String("title").NotEmpty().Comment("合同标题"),
		field.String("party_a").Default("").Comment("甲方"),
		field.String("party_b").Default("").Comment("乙方"),
		field.Float("amount").Default(0).Comment("合同金额"),
		field.Time("start_date").Default(time.Now).Comment("开始日期"),
		field.Time("end_date").Comment("结束日期"),
		field.Enum("status").
			Values("draft", "active", "expired", "terminated").
			Default("draft").
			Comment("状态: draft/active/expired/terminated"),
		field.String("file_url").Default("").Comment("合同文件URL"),
		field.String("remark").Default("").Comment("备注"),
	}
}

// Edges of the Contract.
func (Contract) Edges() []ent.Edge {
	return []ent.Edge{
		edge.From("tenant", Tenant.Type).Ref("contracts").Unique().Required().Field("tenant_id"),
	}
}

// Indexes of the Contract.
func (Contract) Indexes() []ent.Index {
	return []ent.Index{
		index.Fields("tenant_id"),
		index.Fields("status"),
		index.Fields("end_date"),
	}
}
